### DISCUSSION TITLE
Typing the store function

### DISCUSSION THREAD

**Discussion by Hanxi-Yang** [pmndrs/zustand#1500] [Q&A] [answered]
Hello! In the following code, I'm trying to extract the `store` function to its own variable. This is to avoid indentation when you have too many middlewares.
But the type checking is throwing an error because the `store` function is not properly typed. How to type the store function? I didn't find much info on this subject. Would really appreciate it if anyone can point me to the right resource/docs!

```
import create from 'zustand';
import { devtools } from 'zustand/middleware';
import { immer } from 'zustand/middleware/immer';

type Store = {
	foo: string;
	fetchFoo: () => void;
};

const store = (set) => ({
	foo: '',
	async fetchFoo() {
		const response = await Promise.resolve('bar');
		set({ foo: response });
	},
});

const useStore = create<Store>()(immer(devtools(store))); 

export default useStore;
```

**Comment by dai-shi** [comment, marked as answer]
https://github.com/pmndrs/zustand/blob/a94941dce3317a1534e07192705b5ad58b6bf191/docs/guides/typescript.md#slices-pattern

> to avoid indentation

I understand it but avoiding manual typing is easier...

**Reply by Hanxi-Yang** [reply]
thank you!
